internal/webservice/helpers.go

```diff
@@ -0,0 +1,123 @@
+package webservice
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/G-Research/yunikorn-history-server/internal/database/repository"
+)
+
+const (
+	queryParamSubmissionStartTime = "submissionStartTime"
+	queryParamSubmissionEndTime   = "submissionEndTime"
+	queryParamGroups              = "groups"
+	queryParamLimit               = "limit"
+	queryParamOffset              = "offset"
+	queryParamUser                = "user"
+)
+
+func parseApplicationFilters(r *http.Request) (*repository.ApplicationFilters, error) {
+	filters := repository.ApplicationFilters{}
+	user := getUserQueryParam(r)
+	if user != "" {
+		filters.User = &user
+	}
```

Comment:
Do we need to add/set a filter for `groups` here? I don't see any calls to `getGroupsQueryParam(..)` around here.

Reply:
added